src/pages/trade/ui/page.tsx

```diff
@@ -8,49 +8,145 @@ import { TradesTabs } from './trades-tabs';
+          <div className='col-span-2 h-[650px] border-b border-b-other-solidStroke'>
+            <Chart />
+          </div>
+          <RouteTabs />
+          <div className='border-l border-l-other-solidStroke'>
+            <TradesTabs />
+          </div>
+        </div>
       </div>
-
-      <div className='row-start-3 tablet:col-start-2 desktop:row-start-2 desktop:col-start-3 desktop:row-span-2 lg:row-span-1 xl:row-span-2 border-b border-b-other-solidStroke desktop:border-l desktop:border-l-other-solidStroke'>
+      <div className='border-l border-l-other-solidStroke'>
         <FormTabs />
       </div>
-
-      <div className='row-start-4 tablet:row-start-3 lg:row-start-2 lg:col-start-2 xl:row-start-1 xl:row-span-2 border-b border-b-other-solidStroke tablet:border-r tablet:border-r-other-solidStroke lg:border-r-0'>
-        <RouteTabs />
+      <div className='col-span-3 border-t border-t-other-solidStroke'>
+        <HistoryTabs />
       </div>
+    </div>
+  );
+};
 
-      <div className='row-start-2 tablet:col-span-2 desktop:hidden border-b border-b-other-solidStroke'>
+// tablet grid (600px)
+const TabletLayout = () => {
+  return (
+    <div
+      className={cn(sharedStyle, 'hidden tablet:grid desktop:hidden tablet:grid-cols-[1fr,1fr]')}
+    >
+      <div className='col-span-2 border-b border-b-other-solidStroke'>
+        <PairInfo />
+      </div>
+      <div className='col-span-2 border-b border-b-other-solidStroke'>
         <TradesTabs withChart />
       </div>
-      <div className='hidden desktop:block desktop:row-start-3 desktop:col-start-2 lg:col-start-3 desktop:border-b desktop:border-b-other-solidStroke lg:border-l lg:border-l-other-solidStroke'>
-        <TradesTabs />
+      <RouteTabs />
+      <div className='border-l border-l-other-solidStroke'>
+        <FormTabs />
       </div>
-
-      <div className='row-start-5 tablet:row-start-4 tablet:col-span-2 desktop:col-span-3 lg:row-start-3 lg:col-start-1 lg:col-span-2 border-b border-b-other-solidStroke'>
+      <div className='col-span-2 border-t border-t-other-solidStroke'>
         <HistoryTabs />
       </div>
     </div>
   );
 };
+
+// mobile grid (<600px)
+const MobileLayout = () => {
+  return (
+    <div className={cn(sharedStyle, 'hidden mobile:grid tablet:hidden mobile:grid-cols-[1fr]')}>
+      <div className='border-b border-b-other-solidStroke'>
+        <PairInfo />
+      </div>
+      <div className='border-b border-b-other-solidStroke'>
+        <TradesTabs withChart />
+      </div>
+      <div className='border-b border-b-other-solidStroke'>
+        <FormTabs />
+      </div>
+      <div className='border-b border-b-other-solidStroke'>
+        <RouteTabs />
+      </div>
+      <HistoryTabs />
+    </div>
+  );
+};
+
+export const TradePage = () => {
+  return (
+    <>
+      <XlLayout />
+      <LLayout />
+      <DesktopLayout />
+      <TabletLayout />
+      <MobileLayout />
+    </>
+  );
+};
```

Comment:
While it would have been nice to pull this off with pure CSS, I found it kinda unattainable with the numerous layout versions we had.